notebooks/05b Modelling - regression fairness.ipynb

```diff
@@ -185,16 +211,20 @@
    },
    "outputs": [],
    "source": [
-    "model = \"catboost\"\n",
-    "# Generate predictions\n",
-    "preds = np.clip(\n",
-    "    models[model][\"model\"].predict(X_test.drop(columns=sensitive_columns)), 0, None\n",
-    ")\n",
+    "global_mean = fairness_df.LENGTH_OF_STAY.mean()\n",
+    "for feature in sensitive_columns:\n",
+    "    data = pd.DataFrame(columns=[\"subfeature\", \"mean\"])\n",
+    "    # iterate through all subcategories\n",
+    "    for subfeature in fairness_df[feature].unique():\n",
+    "        subset = fairness_df[fairness_df[feature] == subfeature]\n",
+    "        mean_los = subset.LENGTH_OF_STAY.mean()\n",
+    "        df = pd.DataFrame(data=[{\"subfeature\": subfeature, \"mean_los\": mean_los}])\n",
+    "        data = pd.concat([data, df])\n",
     "\n",
-    "# Create a combined data frame\n",
-    "fairness_df = X_test.copy()\n",
-    "fairness_df[\"LENGTH_OF_STAY\"] = y_test\n",
-    "fairness_df[\"LENGTH_OF_STAY_PREDICTED\"] = preds"
+    "    fig = px.bar(data, x=\"subfeature\", y=\"mean_los\", title=feature, orientation=\"v\")\n",
```

Comment:
Up to you, but could be worth considering adding a y-axis label that is more descriptive

Reply:
Agreed and updated